007-design-docs.md

```diff
@@ -0,0 +1,94 @@
+
+
+### How to write a design document
+
+The design documents would live in the related repository in the Strimzi Organization and would match the current code state. Naturally it is hard to allign the design documents and the code behaviour, but it would be the respinsibility of a reviewer to identify if any of the changes are impactful enough to potentially effect the design docs. Due to the below described simplicity of the design docs, if they are found to be out of date, it should be fairly simply to raise a pull request with the behavioural changes documented.
+
+A design document would be split into the following parts:
+
+**Title**
+
+The title of the design document, normally this would pertain to a single component or one larger aspect of the whole of Strimzi. New pages inclusion (but not contents) would be discussed and decided on as part of a proposal if applicable, or during development if the complexity of a component merits a design document as judged by the maintainers.
+
+**Overview, Context and Description**
+
+A high level summary that any user of Strimzi should be able understand and use to decide if it’s useful for them to read the rest of the doc.
+A description of the problem the component/code tackles, why the program/code is necessary, what people need to know to assess the success of the implementation.
+
+**Configuration**
+Describe how users interact with this component or system. I.e. configured via envars, custom resources, REST calls...
+This information will be invaluable to a reader to understand where the configuration points are and how to interact with the system/component.
+
+**Dependencies and Scope**
+Is this intended to be a standalone component, are other components dependent on it, or is it dependent on other components.
+Is this intended for a single Kafka cluster or to interact with multiple.
+
+
+### When is a design document complete
+
+A design document is considered complete when the reviewers and maintainers believe the document to accurately reflect the implementation of the components behaviour. 
+Once approved the document is merged and is understood to be correct, any mistakes or changes to behaviour should be corrected in the form of a pull request alongside the code that changes the behavior.
+
+A lot of implementation issues, problems and compromises are only clear during or after development of a feature or change, so a design doc is recommended to be written alongside the code development, this helps in avoiding the writing of invalid design documents and help hilight potentially problematic elements of implementation or important holes in the testing strategy. 
```

Comment:
"hilight" -> "highlight"

I think given that the scope of these design docs are quite high level it could be argued that it would be valuable to write them up front when creating a new component. I could imagine the document being created as a PR initially and then only merged once the component is complete. It's up to you whether you take on this suggestion, but just thought I'd share my thoughts.